addons/account/models/account_journal.py

```diff
@@ -639,14 +638,22 @@ def _fill_missing_values(self, vals):
             if journal_type in ('cash', 'bank') and not has_loss_account:
                 vals['loss_account_id'] = company.default_cash_difference_expense_account_id.id
 
+        if is_import and not vals.get('code'):
+            code = vals['name'][:5]
+            vals['code'] = code if not codes or code not in codes else self.get_next_bank_cash_default_code(journal_type, company, codes)
+            if not vals['code']:
+                raise UserError(_("Cannot generate an unused journal code. Please change the name for journal %s.", vals['name']))
+
         # === Fill missing refund_sequence ===
         if 'refund_sequence' not in vals:
             vals['refund_sequence'] = vals['type'] in ('sale', 'purchase')
 
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:
-            self._fill_missing_values(vals)
+            # have to keep track of new journal codes when importing
+            codes = [vals['code'] for vals in vals_list if 'code' in vals] if 'import_file' in self.env.context else False
```

Comment:
move it outside the loop?

Reply:
since we update the vals['code'] at each iteration, I don't think I can move it outside :(